nautobot/extras/tests/test_api.py

```diff
@@ -200,14 +200,30 @@ class GitRepositoryTest(APIViewTestCases.APIViewTestCase):
 
     @classmethod
     def setUpTestData(cls):
-        repos = (
-            GitRepository(name="Repo 1", slug="repo-1", remote_url="https://example.com/repo1.git"),
-            GitRepository(name="Repo 2", slug="repo-2", remote_url="https://example.com/repo2.git"),
-            GitRepository(name="Repo 3", slug="repo-3", remote_url="https://example.com/repo3.git"),
+        cls.repos = (
+            GitRepository.objects.create(name="Repo 1", slug="repo-1", remote_url="https://example.com/repo1.git"),
+            GitRepository.objects.create(name="Repo 2", slug="repo-2", remote_url="https://example.com/repo2.git"),
+            GitRepository.objects.create(name="Repo 3", slug="repo-3", remote_url="https://example.com/repo3.git"),
         )
-        for repo in repos:
+        for repo in cls.repos:
             repo.save(trigger_resync=False)
 
+    @skipIf(not rq_worker_running, "RQ worker not running")
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[], JOBS_ROOT=THIS_DIRECTORY)
+    def test_run_git_sync_without_permissions(self):
+        url = reverse("extras-api:gitrepository-run", kwargs={"pk": self.repos[0].id})
+        response = self.client.post(url, format="json", **self.header)
+        self.assertHttpStatus(response, status.HTTP_403_FORBIDDEN)
+
+    @skipIf(not rq_worker_running, "RQ worker not running")
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[], JOBS_ROOT=THIS_DIRECTORY)
+    def test_run_git_sync_with_permissions(self):
+        self.add_permissions("extras.add_gitrepository")
```

Comment:
It may be that we should require both permissions to be able to invoke this endpoint; you can add an explicit check for the change permission:

```python
        if not request.user.has_perm("extras.change_gitrepository"):
            raise PermissionDenied("This user does not have permission to make changes to Git repositories.")
```